File: lab01/ex3/hello.ex2.c
/* 
 * hello.ex2.c 
 * Parallel version using MPI calls
 * Modified from basic version so that workers send back 
 * a message to the master, who prints out a message 
 * for each worker 
 * In addition, the master now sends out two messages to each worker, with 
 * two different tags, and each worker receives the messages 
 * in reverse order. 
 * RLF 10/23/95
*/

#include <stddef.h>
#include <stdlib.h>
#include "mpi.h"
main(int argc, char **argv )
{
	char message1[7], message2[7];
	int i,rank, size, type=99;
        int type2=100;
	MPI_Status status;

	MPI_Init(&argc, &argv);
	MPI_Comm_size(MPI_COMM_WORLD,&size);
	MPI_Comm_rank(MPI_COMM_WORLD, &rank);

	if(rank == 0) {
	   strcpy(message1, "Hello,");
           strcpy(message2, "world");
	   for (i=1; i<size; i++) {
 ===>        (ROTINA MPI)
 ===>        (ROTINA MPI)	   
           }
           printf("node %d : %.7s %.7s\n", rank, message1, message2);
	} 
  	else {
 ===>        (ROTINA MPI)
 ===>        (ROTINA MPI)
          printf("node %d : %.7s , %.7s\n", rank, message1, message2);

          
          }
      	MPI_Finalize();
}

File: MPI/lab01/ex3/hello.ex2.f
C Program hello.ex2.f
C Parallel version using MPI calls.
C Modified from basic version so that workers send back a message to the 
C master, who prints out a message for each worker.  In addition, the 
C master now sends out two messages to each worker, with two different 
C tags, and the worker receives the messages in reverse order.
C RLF 10/11/95

	program hello
	include 'mpif.h'
	integer me,nt,mpierr,tag,status(MPI_STATUS_SIZE)
	integer rank, tag2
	character(6) message1, message2, inmsg1, inmsg2

	call MPI_INIT(mpierr)
	call MPI_COMM_SIZE(MPI_COMM_WORLD,nt,mpierr)
	call MPI_COMM_RANK(MPI_COMM_WORLD,me,mpierr)
        tag = 100
        tag2 = 200

	if(me .eq. 0) then
	  message1 = 'Hello,'
          message2 = 'world'
	  do i=1,nt-1
 ===>       (ROTINA MPI)
 ===>       (ROTINA MPI)
          enddo
          write(6,*)'node',me,':',message1,' ',message2
       else
 ===>       (ROTINA MPI)
 ===>       (ROTINA MPI)      
            write(6,*) 'node', me, ':', inmsg1,' ',inmsg2         
       endif
       CALL MPI_FINALIZE(mpierr)
       end 

